demo/demo.h

```diff
@@ -37,37 +44,207 @@ DECLARE_int32(demo_port);
+  }
+};
+
 
-struct DemoServer {
+class DemoServer {
+ public:
   DemoServer(int port = FLAGS_demo_port) : port_(port) {
+    printf("Preparing to listen on port %d...\n", port_);
     HTTP(port).Register("/ok", [](Request r) { r.connection.SendHTTPResponse("OK\n"); });
     HTTP(port).Register("/uptime", UptimeTracker());
-    HTTP(port).Register("/", State::ClassBoundaries);
-    FileSystem::ScanDir("static", [&port](const std::string& fn) {
+    HTTP(port).Register("/yingyang.svg", State::ClassBoundaries);
+    HTTP(port).Register("/config.json", [](Request r) {
+      // TODO(sompylasar): Use cerealize.
+      r.connection.SendHTTPResponse(
+          "{}\n",
+          HTTPResponseCode::OK, "application/json");
+    });
+    HTTP(port).Register("/layout/data", [](Request r) {
+      std::thread([](Request&& r) {
+                    // Since we are in another thread, need to catch exceptions ourselves.
+                    try {
+                      auto response = r.connection.SendChunkedHTTPResponse(
+                          HTTPResponseCode::OK,
+                          "application/json; charset=utf-8",
+                          {{"Connection", "keep-alive"}, {"Access-Control-Allow-Origin", "*"}});
+                      std::string data;
+                      const double begin = static_cast<double>(Now());
+                      const double t = atof(r.url.query["t"].c_str());
+                      const double end = (t > 0) ? (begin + t * 1e3) : 1e18;
+                      double current;
+                      while ((current = static_cast<double>(Now())) < end) {
+                        std::this_thread::sleep_for(std::chrono::milliseconds(rand() % 100 + 100));
+                        const double x = current;
+                        const double y = sin(5e-3 * (current - begin));
+                        data += Printf("{\"x\":%lf,\"y\":%lf}\n", x, y);
+                        const double f = (rand() % 101) * (rand() % 101) * (rand() % 101) * 1e-6;
+                        const size_t n = static_cast<size_t>(data.length() * f);
+                        if (n) {
+                          response.Send(data.substr(0, n));
+                          data = data.substr(n);
+                        }
+                      }
+                    } catch (const std::exception& e) {
+                      std::cerr << "Exception in data serving thread: " << e.what() << std::endl;
+                    }
+                  },
+                  std::move(r)).detach();
+    });
+    HTTP(port).Register("/layout/meta", [](Request r) {
+      r.connection.SendHTTPResponse(ExampleMeta(),
+                                    "meta",
+                                    HTTPResponseCode::OK,
+                                    "application/json; charset=utf-8",
+                                    {{"Connection", "close"}, {"Access-Control-Allow-Origin", "*"}});
```

Comment:
I think we need to move these ```{"Connection", "close"}``` and ```{"Connection", "keepalive"}``` into Bricks' ```net/api```, for non-chunked and chunked responses respectively. Would that make sense, or are there any possible usecases where it would not?